src/main/scala/org/ergoplatform/settings/Algos.scala

```diff
@@ -17,11 +17,19 @@ object Algos extends ErgoAlgos with scorex.core.utils.ScorexEncoding {
 
   @inline def encode(id: ModifierId): String = encoder.encode(id)
 
-  def blockIdDifficulty(id: Array[Byte]): BigInt = {
-    val blockTarget = BigInt(1, id).ensuring(_ <= Constants.MaxTarget)
-    Constants.MaxTarget / blockTarget
-  }
+  /**
+    * A method to build a Merkle tree over binary objects (leafs of the tree)
+    * @param elements - Merkle tree leafs (byte arrays of arbitrary size)
+    * @return a Merkle tree built over the elements
+    */
+  def merkleTree(elements: Seq[LeafData]): MerkleTree[Digest32] = MerkleTree(elements)(hash)
 
+  /**
+    * A method which is building a Merkle tree over binary objects and returns digest
+    * (256-bits long root hash) of the tree
+    * @param elements - tree leafs
+    * @return 256-bits (32-bytes) long digest of the tree
+    */
   def merkleTreeRoot(elements: Seq[LeafData]): Digest32 =
-    if (elements.isEmpty) emptyMerkleTreeRoot else MerkleTree(elements)(hash).rootHash
+    if (elements.isEmpty) emptyMerkleTreeRoot else merkleTree(elements).rootHash
```

Comment:
How `emptyMerkleTreeRoot` is different from `EmptyRootNode.hash`, which `MerkleTree.rootHash`should return  if `elements` are empty? Also, `emptyMerkleTreeRoot` is a hash of empty byte array and `EmptyRootNode.hash` is `hf.byteArrayToDigest(Array.fill(hf.DigestSize)(0: Byte)).get` (hash of non-empty byte array).

Reply:
Good catch! Made an issue on that: https://github.com/ergoplatform/ergo/issues/1077